disco/disco.go

```diff
@@ -171,24 +221,11 @@ func (d *Disco) discover(hostname svchost.Hostname) (*Host, error) {
 		Path:   discoPath,
 	}
 
-	client := &http.Client{
-		Transport: d.Transport,
-		Timeout:   discoTimeout,
-
-		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			log.Printf("[DEBUG] Service discovery redirected to %s", req.URL)
-			if len(via) > maxRedirects {
-				return errors.New("too many redirects") // this error will never actually be seen
-			}
-			return nil
-		},
-	}
-
-	req := &http.Request{
-		Header: make(http.Header),
-		Method: "GET",
-		URL:    discoURL,
+	req, err := retryablehttp.Get(discoURL.String())
```

Comment:
I think this area of code might not be doing what you intend: 

- `req` in this case is an `*http.Response`.
- The package-level `Get` function (rather than the method on the retryable client) uses a default client, so it's not using the one you set up and stashed in the disco struct.
- Then a `client.Do` call re-does the request you just did, but as best as I can tell it's also not using your retryable client, instead using the inner client you based it on at the start.